src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -20,6 +21,13 @@ public class DataAnnotationsMetadataProvider :
         IDisplayMetadataProvider,
         IValidationMetadataProvider
     {
+        private IStringLocalizerFactory _stringLocalizerFactory { get; set; }
```

Comment:
No need for a property/settability: `private readonly IStringLocalizerFactory _stringLocalizerFactory;`